picard/webservice/__init__.py

```diff
@@ -61,78 +61,7 @@
                                                          PICARD_VERSION_STR)))
 
 
-def escape_lucene_query(text):
-    return re.sub(r'([+\-&|!(){}\[\]\^"~*?:\\/])', r'\\\1', text)
-
-
-def _wrap_xml_metadata(data):
-    return ('<?xml version="1.0" encoding="UTF-8"?>' +
-            '<metadata xmlns="http://musicbrainz.org/ns/mmd-2.0#">%s</metadata>' % data)
-
-
-class XmlNode(object):
-
-    def __init__(self):
-        self.text = ''
-        self.children = {}
-        self.attribs = {}
-
-    def __repr__(self):
-        return repr(self.__dict__)
-
-    def append_child(self, name, node=None):
-        if node is None:
-            node = XmlNode()
-        self.children.setdefault(name, []).append(node)
-        return node
-
-    def __getattr__(self, name):
-        try:
-            return self.children[name]
-        except KeyError:
-            try:
-                return self.attribs[name]
-            except KeyError:
-                raise AttributeError(name)
-
-
-_node_name_re = re.compile('[^a-zA-Z0-9]')
-
-
-def _node_name(n):
-    return _node_name_re.sub('_', n)
-
-
-def _read_xml(stream):
-    document = XmlNode()
-    current_node = document
-    path = []
-
-    while not stream.atEnd():
-        stream.readNext()
-
-        if stream.isStartElement():
-            node = XmlNode()
-            attrs = stream.attributes()
-
-            for i in range(attrs.count()):
-                attr = attrs.at(i)
-                node.attribs[_node_name(attr.name())] = string_(attr.value())
-
-            current_node.append_child(_node_name(stream.name()), node)
-            path.append(current_node)
-            current_node = node
-
-        elif stream.isEndElement():
-            current_node = path.pop()
-
-        elif stream.isCharacters():
-            current_node.text += stream.text()
-
-    return document
-
-
-class XmlWebService(QtCore.QObject):
+class WebService(QtCore.QObject):
```

Comment:
Is it a good idea to provide multiple WebService implementations? Each one would have its own set of queues, which I do find a bit weird...